compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1937,13 +1937,59 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
    TR::Node     *firstChild     = node->getFirstChild();
    TR::Node     *secondChild    = node->getSecondChild();
    bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool isMemRef                = firstChild->getOpCode().isMemoryReference();
+   TR_X86OpCodes opCode, regOpCode, memOpCode, movOpCode;
    intptrj_t    value;
 
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
 
+   // Opcode evaluation for bmul, smul imul and lmul
+   switch (secondChild->getDataType())
+      {
+      case TR::Int8:
+         value = secondChild->getByte();
```

Comment:
You need to check if the node is a constant before you call this API in general. Same for the other constants.

Reply:
Checking for only constant before assigning opCodes will fail all the load cases. So I havent done this. (In other places in the code we have used the API without checking for constants.)

Reply:
I don't see why you cannot do this check - it is wrong to ask a non-const node for its const value